server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -620,6 +620,10 @@ public void skipTranslogRecovery() {
         pendingTranslogRecovery.set(false); // we are good - now we can commit
     }
 
+    public boolean pendingTranslogRecovery() {
```

Comment:
oh, one change, can we make this protected instead? Looks like it is only used by the subclass so just choosing the least visible option.